docs/getting-started/advanced-guide/schema.md

```diff
@@ -0,0 +1,429 @@
+Alternatively, a single overall schema file could be used for Elements and Types. There's also no requirement to have a single Types schemas.
+When you construct your Gaffer graph you must provide all the Schema parts. If provided separately, these will then be merged together to form an internal schema used for the Graph.
+
+When using Java directly, you don't need to use JSON files to store you schemas - although this is still recommended. See the [Java API](#java-api) section for more about interacting with Schemas in Java.
+
+The sections below walkthrough the features of Schemas in detail and explain how to accurately model your data in Gaffer.
+
+
+## Elements schema
+The Elements schema is designed to be a high level document describing what information your Graph contains, i.e. the different kinds of edges and entities and the list of properties associated with each.
+Essentially this part of the schema should just be a list of all the entities and edges in the graph. 
+Edges describe the relationship between a source vertex and a destination vertex. 
+Entities describe a vertex. We use the term "element" to mean either and edge or and entity.
+
+When defining an element we must provide a "group". This is a unique string identifier for each element.
+Groups must be completely unique and cannot be shared between edges and entities.
+
+### Edges
+Edges must have the following:
+
+- `source` - The type of object that will be used as the source vertex in your graph. Needs to be a type defined in the [Types schema](#types).
+- `destination` - Similar to source, it can either be the same or a different type.
+- `directed` - Tells Gaffer if the edge is directed or undirected. Needs to be a type which means true or false, see [Types](#true--false) for more info.
+
+When an Edge is undirected in Gaffer (`directed` is `false`), it is treated as if the relationship was bidirectional and the vertices of the edge do not have an authoritative source and destination thus, the undirected edges `A -- B` and `B -- A` are equal, and therefore will be aggregated together.
+Gaffer will present the undirected edges Vertices in natural ordering, so a client will also see the above edge presented as `A, B`.
+It will be aggregated with any other edge that has the same source and destination and is also undirected.
+For example an undirected Edge that is added, `A -> B`, will be aggregated with another Edge of `B -> A` if they are undirected.
+
+### Entities
+Entities must have a `vertex` field, which is similar to the `source` and `destination` fields on an edge.
+For example, modelling `London -> Paris` requires an entity definition to represent the city and an edge definition to represent the relationship.
+In that example the entity `vertex` field and the edge `source` and `destination` fields would all be of the same type.
```

Comment:
```suggestion
In this example the entity `vertex` field and the edge `source` and `destination` fields would all be of the same type.
```

Reply:
I'm not sure about this. I added lines 38 & 39 to provide an example, but this example is separate from the main example schema discussed on the page. Which is why I wrote 'that' rather than 'this'. Maybe it would be better to rephrase in general?

Reply:
I had another read and I am still not sure - I think maybe rephrase? I feel the 'that' makes it less clear that you mean the example in the previous sentence.

Reply:
I've reworded this to refer to the previous example to avoid any confusion.